### ISSUE TITLE
[Bug] [iOS] responseType arraybuffer doesn't work for bad status code

### ISSUE THREAD

**Issue by millerg6711** [silkimen/cordova-plugin-advanced-http#407]
**Describe the bug**
If you have the responseType set to "arraybuffer", and a bad server response code is returned... an error will be thrown and the response data will be parsed as a string rather than raw data. Whereas if you have a successful response- the data will be parsed as an arraybuffer as expected.

**System info**
 - affected HTTP plugin version: 3.0.1
 - affected platform(s) and version(s): iOS 14.4.1
 - affected device(s): iPhone 10
 - cordova version: 6.12.3
 - cordova platform version(s): ios 6.1.1

**Minimum viable code to reproduce**
e.g.:
```js
try {
   // server returns 401 status code and raw data
   await this.http.sendRequest(..., {..., responseType: "arraybuffer" });
} catch (e) {
   console.log(e.error);
}
```

**Comment by silkimen** [comment]
Yes, that's right, but this is intended functionality. See readme: https://github.com/silkimen/cordova-plugin-advanced-http#failure The reason is that most APIs do send error strings when something went wrong and the `error` property is meant to be an error message (string not ArrayBuffer).

**Comment by millerg6711** [comment]
> Yes, that's right, but this is intended functionality. See readme: https://github.com/silkimen/cordova-plugin-advanced-http#failure The reason is that most APIs do send error strings when something went wrong and the `error` property is meant to be an error message (string not ArrayBuffer).

I don't have access to the api I'm trying to reach. Some apis return protobufs which can't be encoded as strings. What should I do as a workaround?

**Comment by silkimen** [comment]
For the time being, there is no workaround for your problem. If you want to have an option for configuring the behavior, you can open a new feature request.
